src/circuit/gadget/ecc/chip.rs

```diff
@@ -254,15 +254,21 @@ impl EccChip {
         let mul_overflow =
             mul::overflow::Config::configure(meta, range_check, advices[6..9].try_into().unwrap());
         let mul_complete = mul::complete::Config::configure(meta, advices[9], add);
+        let mul_hi = mul::incomplete::Config::configure(
+            meta, advices[9], advices[3], advices[0], advices[1], advices[4], advices[5],
+        );
+        let mul_lo = mul::incomplete::Config::configure(
+            meta, advices[6], advices[7], advices[0], advices[1], advices[8], advices[2],
+        );
```

Comment:
And then the third commit refactors `mul::incomplete::Config` last. I think we can probably eliminate the constraint reordering entirely if we:
- Reorder the commits to refactor `mul::incomplete`, `mul::complete`, `mul::overflow` in that order.
- Reorder the variables above to be `mul_hi, mul_lo, mul_complete, mul_overflow`.